src/push-state-tree.js

```diff
@@ -268,31 +225,69 @@ function PushStateTree(options) {
 
       uri = this.getUri(url);
 
+      // Update oldURI
+      oldURI = cachedUri.uri;
       // Update cache
       cachedUri.url = url;
       cachedUri.basePath = basePath;
       cachedUri.uri = uri;
 
       // Expose DOM Attribute
-      if (this.getAttribute('uri') !== uri) {
+      if (this.getAttribute('uri') != uri) {
         this.setAttribute('uri', uri);
       }
 
       return uri;
-    },
-    configurable: true
+    }
   });
 
-  //TODO: Make it represent internal baseURI changes from proposal #47
-  proxyReadOnlyProperty(this, 'length', internalHistory);
+  let length = 0;
+  Object.defineProperty(this, 'length', {
+    get() {
+      return length;
+    }
+  });
 
+  let path = '';
   Object.defineProperty(this, 'isPathValid', {
     get() {
-      var uri = internalHistory.last().url;
-      return !this.basePath || uri.indexOf(this.basePath) === 0;
+      return !basePath || path.indexOf(basePath) === 0;
     }
   });
 
+  Object.defineProperty(this, 'path', {
+    get() {
+      return path;
+    },
+    set(value) {
+      if (holdDispatch || value == path || typeof path != 'string') return;
```

Comment:
Wouldn't be better to throw an error instead of just returning if path is not a string?
This can trick the developper into thinking that the value was set when it was not, making a bit harder to debug.